src/usr/local/buildpack/utils/cache.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# will attempt to download the file at the given url and stores it in the cache.
+# If this file already exists, will return the cached version
+# The cache will only used if BUILDPACK_CACHE_DIR is set
+# First argument is the url, second one is the filename (optional)
+function get_from_url () {
+  local url=${1}
+  check url true
+
+  local name
+  name=${2:-$(basename "${url}")}
+
+  if [ -n "${BUILDPACK_CACHE_DIR}" ] && [ -e "${BUILDPACK_CACHE_DIR}/${name}" ]; then
+      # file in cache
+      echo "Found file in cache: ${BUILDPACK_CACHE_DIR}/${name}" >&2
+      echo "${BUILDPACK_CACHE_DIR}/${name}"
+  else
+    # cache disabled or not in cache
+    download_file "${url}" "${name}"
+  fi
+}
+
+# Will download the file into the cache folder and returns the path
+# If the cache is not enabled it will download it to a temp folder
+# The second argument will be the filename if given
+function download_file () {
+  local url=${1}
+  check url true
+
+  local name
+  name=${2:-$(basename "${url}")}
+
+  local temp_folder=${BUILDPACK_CACHE_DIR:-$(mktemp -u)}
```

Comment:
We need to cleanup this when not using cache, as we don't want to have the downloads for prebuild images (eg python or node)

Reply:
We do, my initial thought was to return the path and the user is responsible for cleanup.
We can introduce a general cache dir that we do not need to clean up like `/opt/buildpack/cache/` for downloaded tools when the cache is disabled. How does that sound?

Reply:
Currently we use `/tmp` which is cleaned up by `install-tool` and others. So can we use that as cache when no explicit caching is enabled?

Reply:
`/tmp` is fine with me for now. 
We still need to have the discussion if we want to move the install folder alltogether but that can be done at a later stage.